dask/annotation.py

```diff
@@ -0,0 +1,22 @@
+import functools
+
+
+def annotate_func(func: callable, annotation: dict) -> callable:
+    ret = functools.partial(func)
+    if not hasattr(ret, "_dask_annotation_dict"):
+        ret._dask_annotation_dict = {}
+        if hasattr(func, "__name__"):
```

Comment:
Can you use `functools.update_wrapper` here?

Reply:
Good idea, will use `functools.update_wrapper` instead!